server/autolinkplugin/config.go

```diff
@@ -14,24 +14,34 @@ import (
 type Config struct {
 	EnableAdminCommand bool
 	EnableOnUpdate     bool
+	PluginAdmins       string
 	Links              []autolink.Autolink
+
+	// AdminUserIds is a set of UserIds that are permitted to perform
+	// administrative operations on the plugin configuration (i.e. plugin
+	// admins). On each configuration change the contents of PluginAdmins
+	// config field is parsed into this field.
+	AdminUserIds map[string]struct{}
```

Comment:
Since the number of admin users is likely to be low, I would maybe skip the complexity here and just have the check use the string directly instead of pre-processing it. eg: https://github.com/mattermost/mattermost-plugin-aws-SNS/blob/master/server/plugin.go#L348

Reply:
Pre-parsing UserIds on config change [1] allows us to catch malformed UserIds and log them so that users can debug situations where e.g. `X` does not have access without resorting to digging into the code. Additionally the `map[string]struct{}` + `if _, ok := foo[x]; !ok` is a common idiom in Go for a set and set membership tests, at least I find it pretty readable.

Let me know if you feel strongly about it and I will change it as requested, if not - it would be great if we could leave it as-is.

[1] https://github.com/mattermost/mattermost-plugin-autolink/pull/93/files/bb32075fc5b2f4e01ff880d55e752741bdb2ef0b#diff-45a5ec4d780fb965962fe72d26e03952R139-R144

Reply:
Ok. Sounds good.